### PR TITLE
Added Tutorials in resources collections

### PR DESCRIPTION
**Pull request by YashKamboj**
Signed-off-by: YashKamboj <75125203+YashKamboj@users.noreply.github.com>

**Description**
Added Tutorials in resources collections
This PR fixes #

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 95d01370a3e7bf69ba3f349f237b3e9e90455d00 at: https://61fa6e758d3b5a00634e6b90--layer5ng.netlify.app

**Review by Nikhil-Ladha — CHANGES_REQUESTED** [review]
Never have file/folder/image name with spaces! Please update them to use a hyphen (-) instead.
Also, no need for the folder/image names to be capitalized.

**Comment by l5io** [comment]
🚀 Preview for commit 19ea31bca6cc1c25acd56e885029219908ace84a at: https://61fba455804d9464a0c3da7a--layer5ng.netlify.app

File: src/collections/tutorials/Resources.style.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import styled from "styled-components";
+export const ResourcesWrapper = styled.div`
```
Comment (debo19) [inline src/collections/tutorials/Resources.style.js:2]: This file is not required.

**Review by debo19 — COMMENTED** [review]
Since you've created an entirely new collection, the "Related Resources" is not configured for each tutorial. Now instead of doing that all over again, I think the better way will be to just include a `tutorials` folder inside the `resources` folder itself. Check out the `recorded webinar` inside `resources`. That will unite this with other resources.

**Comment by YashKamboj** [comment]
@debo19 can you help me , i can't find why its failing, didn't changed much which may cause this error.

**Review by warunicorn19 — CHANGES_REQUESTED** [review]

File: src/collections/resources/tutorials/contributing-to-meshery-API-swagger-documentation/index.mdx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+author: Piyush Singariya
+type: Tutorial
+product: Meshery
+technology: API
+featured: false
+published: true
+resource: true
+---
+
+import { ResourcesWrapper } from "../../Resources.style.js";
```
Comment (warunicorn19) [inline src/collections/resources/tutorials/contributing-to-meshery-API-swagger-documentation/index.mdx:14]:
```suggestion
import { ResourcesWrapper } from "../Resources.style.js";
```

Comment (warunicorn19) [reply src/collections/resources/tutorials/contributing-to-meshery-API-swagger-documentation/index.mdx:14]: this stylesheet is right outside the resource folder

File: src/sections/Resources/Resources-grid/filters.js [hunk 3]
```diff
@@ -14,31 +14,31 @@ const Navigation = (props) => {
     graphql`
             query allFilters {
                 type: allMdx(
-                    filter: { fields: { collection: { in: ["blog", "resources", "news", "events"] } }, frontmatter: { published: { eq: true } , resource: { eq: true} }}
+                    filter: { fields: { collection: { in: ["blog", "resources", "news", "events", "tutorials"] } }, frontmatter: { published: { eq: true } , resource: { eq: true} }}
```
Comment (debo19) [inline src/sections/Resources/Resources-grid/filters.js:17]:
```suggestion
                    filter: { fields: { collection: { in: ["blog", "resources", "news", "events"] } }, frontmatter: { published: { eq: true } , resource: { eq: true} }}
```
This won't be required anymore.

File: src/sections/Resources/Resources-grid/filters.js [hunk 4]
```diff
@@ -14,31 +14,31 @@ const Navigation = (props) => {
...
                 ){
                     group(field: frontmatter___type) {
                         fieldValue
                         totalCount
                     }
                 }
                 product: allMdx(
-                  filter: { fields: { collection: { in: ["blog", "resources", "news", "events"] } }, frontmatter: { published: { eq: true } , resource: { eq: true} }}
+                  filter: { fields: { collection: { in: ["blog", "resources", "news", "events", "tutorials"] } }, frontmatter: { published: { eq: true } , resource: { eq: true} }}
```
Comment (debo19) [inline src/sections/Resources/Resources-grid/filters.js:25]:
```suggestion
                  filter: { fields: { collection: { in: ["blog", "resources", "news", "events"] } }, frontmatter: { published: { eq: true } , resource: { eq: true} }}
```

File: gatsby-config.js [hunk 5]
```diff
@@ -415,7 +415,7 @@ module.exports = {
     {
       resolve: "gatsby-source-filesystem",
       options: {
-        path: `${__dirname}/src/collections/tutorials`,
+        path: `${__dirname}/src/collections/resources/tutorials`,
```
[the remark this answers is not in the document]
Comment (debo19) [reply gatsby-config.js:418]: No changes will be needed in the gatsby-config now.

File: src/sections/Resources/Resources-grid/DataWrapper.js [hunk 6]
```diff
@@ -11,7 +11,7 @@ const DataWrapper = (WrappedComponent) => {
             allMdx(
                 sort: {fields: [frontmatter___date], order: DESC}
                 filter: {
-                    fields: { collection: { in: ["blog", "resources", "news", "events"] } }
+                    fields: { collection: { in: ["blog", "resources", "news", "events", "tutorials"] } }
```
Comment (debo19) [inline src/sections/Resources/Resources-grid/DataWrapper.js:14]:
```suggestion
                    fields: { collection: { in: ["blog", "resources", "news", "events"] } }
```

**Review by debo19 — CHANGES_REQUESTED** [review]

**Comment by YashKamboj** [comment]
@debo19 Do you know why is the build failing can't find the reason, my development server is also failing when opening any resources with related resources.

**Comment by asubedy** [comment]
@YashKamboj add a `tags` field after category field that should solve the error. Like this:
image

**Comment by debo19** [comment]
> @YashKamboj add a `tags` field after category field that should solve the error. Like this: image

Yes :grimacing: That isn't very obvious right now :sweat: I will go ahead and create a resource template and add a brief note on each of these fields that are required.

**Comment by YashKamboj** [comment]
@asubedy Thnx it worked .

**Comment by l5io** [comment]
🚀 Preview for commit 77f29c6a4e1d384abf19d80314f4acc28f16eb58 at: https://61fd1075fe40ddb485ac7174--layer5ng.netlify.app

File: src/collections/resources/tutorials/functional-testing-with-cypress-in-meshery-ui/index.mdx [hunk 7]
```diff
@@ -0,0 +1,109 @@
+---
+title: "Functional Testing with Cypress in Meshery UI"
```
Comment (debo19) [inline src/collections/resources/tutorials/functional-testing-with-cypress-in-meshery-ui/index.mdx:2]: This is a blog post right - https://layer5.io/blog/meshery/functional-testing-with-cypress-in-meshery-ui
We can remove this from tutorials.

File: src/collections/resources/tutorials/introduction-to-all-layer5-repositories/index.mdx [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+featured: false
+published: true
+resource: true
+---
+
+import { ResourcesWrapper } from "../../Resources.style.js";
+
+<ResourcesWrapper> 
+        
+        <iframe src="https://docs.google.com/document/d/e/2PACX-1vQEnQ6m6uvAYPFLlBdKQLoB36YKrHh3A7EBSUVfMkSleoDagO4sa2WYg0tDQk51Su6gloAwHUhijO6W/pub?embedded=true"
```
Comment (debo19) [inline src/collections/resources/tutorials/introduction-to-all-layer5-repositories/index.mdx:19]: This doesn't look good inside an `iframe`. Instead, just add a brief note to check out the `Repository Overview` and link to https://layer5.io/community/handbook/repository-overview

**Review by debo19 — CHANGES_REQUESTED** [review]

**Review by debo19 — COMMENTED** [review]
Try adding a small heading before each slide or recording. ("Tutorial Slides" "Tutorial Recording")

**Comment by l5io** [comment]
🚀 Preview for commit 94164485e65df49f15977737463df141f94a1ae3 at: https://61fd641004d7f04e6fa8dffc--layer5ng.netlify.app

**Comment by warunicorn19** [comment]
slide deck on the first tutorial is broken

File: src/collections/resources/tutorials/introduction-to-all-layer5-repositories/index.mdx (hunk 8 above)
Comment (debo19) [reply src/collections/resources/tutorials/introduction-to-all-layer5-repositories/index.mdx]: @YashKamboj After the recording, you can just add a button like we have for any [event](https://layer5.io/community/events/project-office-hours-service-mesh-performance-at-kubecon-na-2021) to say "Check out the Layer5 Repository Overview"

File: src/collections/resources/tutorials/tutorial-on-contributing-to-layer5-and-working-with-git.mdx [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+type: Tutorial
+featured: false
+published: true
+resource: true
+---
+
+import { ResourcesWrapper } from "../Resources.style.js";
+
+<ResourcesWrapper> 
+      <h1> Tutorial Slides </h1>
```
Comment (debo19) [inline src/collections/resources/tutorials/tutorial-on-contributing-to-layer5-and-working-with-git.mdx:18]:
```suggestion
      <h3> Tutorial Slides </h3>
```
`<h3>` will be better here.

Comment (debo19) [reply src/collections/resources/tutorials/tutorial-on-contributing-to-layer5-and-working-with-git.mdx:18]: Not just this, the rest of the headings

**Comment by YashKamboj** [comment]
@warunicorn19 working fine for me, can you check again

**Comment by l5io** [comment]
🚀 Preview for commit 726f22d669b9791817c5b017c78e3c23b423f8fe at: https://61feb31bf95e0155df905517--layer5ng.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 375ccb3b6d6498b24df1b862fbbca6829128c8e0 at: https://61feb804f8236b57b571f06d--layer5ng.netlify.app

File: src/collections/resources/tutorials/beginner-guide-to-contributing-to-meshery-and-mesheryctl.mdx [hunk 10]
```diff
@@ -0,0 +1,27 @@
+---
+title: "Beginner's guide to contributing to Meshery and mesheryctl"
+thumbnail: ../../../assets/images/tutorials/intro-meshery-ctl.png
+category: Service Mesh
+tags: 
+  - Meshery
+  - Tutorial
+author: Navendu Pottekkat
```
Comment (debo19) [inline src/collections/resources/tutorials/beginner-guide-to-contributing-to-meshery-and-mesheryctl.mdx:8]:
```suggestion
```
We avoid adding authors to resources. Please make this change in the other files as well.

File: src/collections/resources/tutorials/introduction-to-meshey-ctl.mdx [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+
+import { ResourcesWrapper } from "../Resources.style.js";
+
+<ResourcesWrapper> 
+      <h3> Tutorial Slides </h3>
+        <iframe	 src="https://docs.google.com/presentation/d/1LxhzJhUs9-Hc9mwlKVUMlDAVdH-QWLvU1KWxVuRbSHg/embed?start=false&loop=false&delayms=3000"	
+              frameBorder="0" allowFullScreen width="100%" height="500">	
+        </iframe>
+      <h3> Tutorial Recording </h3>
+        <iframe className="iframe" src="https://www.youtube.com/embed/wK7Q-zbJ3gQ&t=3s"
```
Comment (debo19) [inline src/collections/resources/tutorials/introduction-to-meshey-ctl.mdx:23]: This links seems to be broken for me. Can you check this once?

File: src/collections/resources/tutorials/beginner-guide-to-contributing-to-meshery-and-mesheryctl.mdx [hunk 12]
```diff
@@ -0,0 +1,27 @@
+---
+title: "Beginner's guide to contributing to Meshery and mesheryctl"
+thumbnail: ../../../assets/images/tutorials/intro-meshery-ctl.png
```
Comment (debo19) [inline src/collections/resources/tutorials/beginner-guide-to-contributing-to-meshery-and-mesheryctl.mdx:3]: I think you need to change the thumbnail for this one. You seem to have added two duplicate images for [this](https://61feb804f8236b57b571f06d--layer5ng.netlify.app/resources/service-mesh/beginners-guide-to-contributing-to-meshery-and-mesheryctl) and [Intro to mesheryctl](https://61feb804f8236b57b571f06d--layer5ng.netlify.app/resources/service-mesh/an-introduction-to-mesheryctl)

**Review by debo19 — CHANGES_REQUESTED** [review]

**Comment by l5io** [comment]
🚀 Preview for commit bc76432d2b1601ef7f595b7fef349463c345a278 at: https://620008fee12dde4653326457--layer5ng.netlify.app

**Comment by warunicorn19** [comment]
@Nikhil-Ladha everytime I try to see the preview I get this.
Screenshot 2022-02-06 at 1 18 18 PM
Do you think this might be happening becaue of the `gatsby-robots-txt` ?

**Comment by Nikhil-Ladha** [comment]
@warunicorn19 possibly. As, I see the robots file js also not what I intended it to be and I think that's because of the way we create previews.
I will send a PR to revert the changes soon.
Though, I am not seeing such message while opening the preview.

**Comment by leecalcote** [comment]
FWIW: Whether in Safari or Chrome, I don't see these messages either.

**Comment by Nikhil-Ladha** [comment]
@YashKamboj some conflicts!

**Review by debo19 — APPROVED** [review]
LGTM, thanks @YashKamboj 🙌

**Review by leecalcote — CHANGES_REQUESTED** [review]
@YashKamboj will you be so kind as to ensure that all slide deck cover images are the same size? They are all coming from the same source: a Google Slide. Let's ensure that the fit to the size of the resources cards precisely and are of the exact same size.

**Pull request merged**